packages/yoastseo/spec/languageProcessing/researches/keywordCountSpec.js

```diff
@@ -25,221 +26,779 @@ const buildMorphologyMockResearcher = function( keyphraseForms ) {
+					"key</yoastmark> and another <yoastmark class='yoast-text-mark'>key</yoastmark>) and one <yoastmark " +
+					"class='yoast-text-mark'>word</yoastmark>.",
+			original: "A string with three keys (key and another key) and one word.",
+			position: { endOffset: 32, startOffset: 29 } } ),
+			new Mark( { marked: "A string with three <yoastmark class='yoast-text-mark'>keys</yoastmark> (<yoastmark class='yoast-text-mark'>" +
+					"key</yoastmark> and another <yoastmark class='yoast-text-mark'>key</yoastmark>) and one <yoastmark " +
+					"class='yoast-text-mark'>word</yoastmark>.",
+			original: "A string with three keys (key and another key) and one word.",
+			position: { endOffset: 48, startOffset: 45 } } ),
+			new Mark( { marked: "A string with three <yoastmark class='yoast-text-mark'>keys</yoastmark> (<yoastmark class='yoast-text-mark'>" +
+					"key</yoastmark> and another <yoastmark class='yoast-text-mark'>key</yoastmark>) and one <yoastmark " +
+					"class='yoast-text-mark'>word</yoastmark>.",
+			original: "A string with three keys (key and another key) and one word.",
+			position: { endOffset: 62, startOffset: 58 } } ) ],
+		skip: false,
+	},
+	{
+		description: "matches both singular and reduplicated plural form of the keyword in Indonesian, " +
+			"the plural should be counted as one occurrence",
+		paper: new Paper( "<p>Lorem ipsum dolor sit amet, consectetur keyword-keyword, keyword adipiscing elit.</p>",
+			{ locale: "id_ID", keyword: "keyword" } ),
+		keyphraseForms: [ [ "keyword", "keyword-keyword" ] ],
 		expectedCount: 2,
-		expectedMarkings: [ new Mark( { marked: "A string of text with a <yoastmark class='yoast-text-mark'>keyword</yoastmark> " +
-				"and multiple <yoastmark class='yoast-text-mark'>keywords</yoastmark> in it.",
-		original: "A string of text with a keyword and multiple keywords in it." } ) ],
-
+		expectedMarkings: [
+			new Mark( {
+				marked: "Lorem ipsum dolor sit amet, consectetur <yoastmark class='yoast-text-mark'>keyword-keyword</yoastmark>," +
+					" <yoastmark class='yoast-text-mark'>keyword</yoastmark> adipiscing elit.",
+				original: "Lorem ipsum dolor sit amet, consectetur keyword-keyword, keyword adipiscing elit.",
+				position: { endOffset: 58, startOffset: 43 } } ),
+			new Mark( {
+				marked: "Lorem ipsum dolor sit amet, consectetur <yoastmark class='yoast-text-mark'>keyword-keyword</yoastmark>," +
+					" <yoastmark class='yoast-text-mark'>keyword</yoastmark> adipiscing elit.",
+				original: "Lorem ipsum dolor sit amet, consectetur keyword-keyword, keyword adipiscing elit.",
+				position: { endOffset: 67, startOffset: 60 } } ) ],
+		skip: false,
 	},
 	{
-		description: "counts a string with a keyword with a '-' in it",
-		paper: new Paper( "A string with a key-word.", { keyword: "key-word" } ),
-		keyphraseForms: [ [ "key-word", "key-words" ] ],
+		description: "matches both reduplicated keyphrase separated by '-' as two occurrence in English",
```

Comment:
I don't know if this would happen in real life in English 😄 But this case is added to highlight the difference in behavior between Indonesian and non-Indonesian.